### DISCUSSION TITLE
Hello packet multicast problem

### DISCUSSION THREAD

**Discussion by 03psb** [ANLAB-KAIST/KENSv3#245] [Project Q&A 3. Internet Layer (IP/Routing)] [answered]
Hi, I'm working on sending and receiving hello packet. Since the packet is sent as multicast IP, the packet can be reached to multiple destinations
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/98043513/92fa28d3-e2e7-4fc5-9dec-61596905ee85)
but as you can see in this TestRoutingOSPFCustom1
, I found that each hello packet is only destined to one destination, although they should reach multiple destinations.
I'm having hard time figuring out why, so I hope someone can help me with this problem.
below is wireshark capture of one of the produced packet in TestRoutingOSPFCustom1
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/98043513/769d7577-5dc5-47b7-852d-2d8c3aaaa891)

**Comment by 03psb** [comment]
I recognized the first image is actually not from Custom1 example, anyway the point remains same

**Comment by Pusnow** [comment, marked as answer]
> , I found that each hello packet is only destined to one destination, although they should reach multiple destinations.

Currently, each subnet has only two routers, so each multicast packet only reaches one routers.
To send the packet to every neighbor routers, you should repeat `sendPacket` for every interfaces by changing source IP address.
